Storage/StorageTests/CoreData/CoreDataManagerTests.swift

```diff
@@ -70,28 +70,77 @@ final class CoreDataManagerTests: XCTestCase {
         XCTAssertEqual(viewContext.countObjects(ofType: ShippingLine.self), 0)
     }
 
-    func test_saving_derived_storage_while_resetting_CoreData_does_not_save_data() throws {
+    func test_performAndSave_executes_changes_in_background_then_updates_viewContext() throws {
         // Arrange
-        let manager = CoreDataManager(name: storageIdentifier, crashLogger: MockCrashLogger())
-        manager.reset()
+        let modelsInventory = try makeModelsInventory()
+        let manager = try makeManager(using: modelsInventory, deletingExistingStoreFiles: true)
         let viewContext = try XCTUnwrap(manager.viewStorage as? NSManagedObjectContext)
-        let derivedContext = try XCTUnwrap(manager.writerDerivedStorage as? NSManagedObjectContext)
+        XCTAssertEqual(viewContext.countObjects(ofType: Account.self), 0)
 
         // Action
-        waitForExpectation(count: 2) { expectation in
-            derivedContext.perform {
-                _ = derivedContext.insertNewObject(ofType: ShippingLine.self)
-            }
-            manager.saveDerivedType(derivedStorage: derivedContext, {
-                XCTAssertEqual(viewContext.countObjects(ofType: ShippingLine.self), 1)
+        waitForExpectation(count: 1) { expectation in
+            manager.performAndSave({ storage in
+                XCTAssertFalse(Thread.current.isMainThread)
```

Comment:
There is no context here of why `Thread.current.isMainThread)` should be false, I think we should either add a message to the assertion, or update the test name, or declare it somewhere outside the expectation more clearly (eg: let expectedThread = ... )

Reply:
Good point! Added assert messages in f0c461e12fec6d6d306b557ffa1309312b621293.